services/admin/src/server.ts

```diff
@@ -293,6 +303,146 @@ export function buildApp({ store }: { store: Store }): Application {
+      const { electionId, writeInId } = request.params;
+
+      const electionRecord = store.getElection(electionId);
+      if (!electionRecord) {
+        response.status(400).json({
+          status: 'error',
+          errors: [
+            {
+              type: 'invalid-value',
+              message: `invalid election Id`,
+            },
+          ],
+        });
+        return;
+      }
+      const { election } = electionRecord.electionDefinition;
+      const castVoteRecordData = store.getCastVoteRecordForWriteIn(writeInId);
+      if (!castVoteRecordData) {
+        response.status(400).json({
+          status: 'error',
+          errors: [
+            {
+              type: 'invalid-value',
+              message: `invalid write in Id`,
+            },
+          ],
+        });
+        return;
+      }
+      const { contestId, optionId, cvr } = castVoteRecordData;
+
+      try {
+        const ballotStyle = getBallotStyle({
+          ballotStyleId: cvr._ballotStyleId,
+          election,
+        });
+        if (
+          // eslint-disable-next-line no-underscore-dangle
+          cvr._layouts === undefined ||
+          // eslint-disable-next-line no-underscore-dangle
+          cvr._ballotImages === undefined
+        ) {
+          response.json([]); // The CVR does not have ballot images.
+          return;
+        }
+        if (!ballotStyle) {
+          throw new Error('unexpected types');
+        }
+        const allContestIdsForBallotStyle = getContests({
+          ballotStyle,
+          election,
+        }).map((c) => c.id);
+        const [layouts, ballotImages] = [
+          // eslint-disable-next-line no-underscore-dangle
+          ...zip(cvr._layouts, cvr._ballotImages),
+        ]
+          .sort(([a], [b]) => a.metadata.pageNumber - b.metadata.pageNumber)
+          .reduce<[BallotPageLayout[], InlineBallotImage[]]>(
+            ([layoutsAcc, ballotImagesAcc], [layout, ballotImage]) => [
+              [...layoutsAcc, layout],
+              [...ballotImagesAcc, ballotImage],
+            ],
+            [[], []]
+          );
+        let contestIdx = allContestIdsForBallotStyle.indexOf(contestId);
+        let currentLayoutOptionIdx = 0;
+        let currentLayout = layouts[currentLayoutOptionIdx];
+        while (currentLayout && contestIdx >= currentLayout.contests.length) {
+          currentLayoutOptionIdx += 1;
+          currentLayout = layouts[currentLayoutOptionIdx];
+          if (!currentLayout) {
+            throw new Error('unexpected types');
+          }
+          contestIdx -= currentLayout.contests.length;
+        }
+        currentLayout = layouts[currentLayoutOptionIdx];
+        if (!currentLayout) {
+          throw new Error('unexpected types');
+        }
+        const contestLayout =
+          layouts[currentLayoutOptionIdx]?.contests[contestIdx];
+
+        // Options are laid out from the bottom up, so we reverse write-ins to get the correct bounds
```

Comment:
This is only true for NH ballots. If we merge this, I think we need a comment or link to an issue on this endpoint saying what needs to change in the future.

Reply:
https://github.com/votingworks/vxsuite/issues/2564